packages/firestore/lite/test/integration.test.ts

```diff
@@ -832,4 +836,134 @@ describe('Query', () => {
       }
     );
   });
+
+  it('supports collection groups', () => {
+    return withTestCollection(async collRef => {
+      const collectionGroupId = `${collRef.id}group`;
+
+      const fooDoc = doc(
+        collRef.firestore,
+        `${collRef.id}/foo/${collectionGroupId}/doc1`
+      );
+      const barDoc = doc(
+        collRef.firestore,
+        `${collRef.id}/bar/baz/boo/${collectionGroupId}/doc2`
+      );
+      await setDoc(fooDoc, { foo: 1 });
+      await setDoc(barDoc, { bar: 1 });
+
+      const query = collectionGroup(collRef.firestore, collectionGroupId);
+      const result = await getQuery(query);
+
+      verifyResults(result, { bar: 1 }, { foo: 1 });
+    });
+  });
+
+  it('validates collection groups', () => {
+    return withTestDb(firestore => {
+      expect(() => collectionGroup(firestore, '')).to.throw(
+        'Function collectionGroup() requires its first argument to be of type non-empty string, but it was: ""'
+      );
+      expect(() => collectionGroup(firestore, '/')).to.throw(
+        "Invalid collection ID '/' passed to function collectionGroup(). Collection IDs must not contain '/'."
+      );
+    });
+  });
+});
+
+describe('equality', () => {
+  it('for collection references', () => {
+    return withTestDb(firestore => {
+      const coll1a = collection(firestore, 'a');
+      const coll1b = parent(doc(firestore, 'a/b'));
+      const coll2 = collection(firestore, 'c');
+
+      expect(refEqual(coll1a, coll1b)).to.be.true;
+      expect(refEqual(coll1a, coll2)).to.be.false;
+
+      const coll1c = collection(firestore, 'a').withConverter({
+        toFirestore: data => data as firestore.DocumentData,
+        fromFirestore: snap => snap.data()
+      });
+      expect(refEqual(coll1a, coll1c)).to.be.false;
+
+      expect(refEqual(coll1a, doc(firestore, 'a/b'))).to.be.false;
+    });
+  });
+
+  it('for document references', () => {
+    return withTestDb(firestore => {
+      const doc1a = doc(firestore, 'a/b');
+      const doc1b = doc(collection(firestore, 'a'), 'b');
+      const doc2 = doc(firestore, 'a/c');
+
+      expect(refEqual(doc1a, doc1b)).to.be.true;
+      expect(refEqual(doc1a, doc2)).to.be.false;
+
+      const doc1c = collection(firestore, 'a').withConverter({
+        toFirestore: data => data as firestore.DocumentData,
+        fromFirestore: snap => snap.data()
+      });
+      expect(refEqual(doc1a, doc1c)).to.be.false;
+
+      expect(refEqual(doc1a, collection(firestore, 'a'))).to.be.false;
+    });
+  });
+
+  it('for queries', () => {
+    return withTestCollectionAndInitialData(
```

Comment:
is there a reason these test helpers can't live in a beforeEach() block?

Reply:
These helper set up different state each time and make the state visible in the test code. This also matches existing Firestore integration tests.